pkg/compact/downsample/downsample_test.go

```diff
@@ -266,6 +267,40 @@ func TestCounterSeriesIterator(t *testing.T) {
 	testutil.Equals(t, exp, res)
 }
 
+func TestCounterSeriesIteratorSeek(t *testing.T) {
+	defer leaktest.CheckTimeout(t, 10*time.Second)()
+
+	chunks := [][]sample{
+		{{100, 10}, {200, 20}, {300, 10}, {400, 20}, {400, 5}},
+	}
+	exp := []sample{
+		{100, 10}, {200, 20}, {300, 30}, {400, 40},
+	}
+
+	var its []chunkenc.Iterator
+	for _, c := range chunks {
+		its = append(its, newSampleIterator(c))
+	}
+
+	x := NewCounterSeriesIterator(its...)
+	var it *storage.BufferedSeriesIterator
+	it = storage.NewBuffer(x, 500)
+	ok := it.Seek(500)
+	if !ok {
+		if it.Err() != nil {
```

Comment:
I think instead of all of this (and `if !ok`) we need just
```
testutil.True(t, !ok, "...") // since seek 500 should return false
testutil.Ok(t, it.Err())
```